daprdocs/content/en/developing-applications/middleware/supported-middleware/middleware-sentinel.md

```diff
@@ -0,0 +1,85 @@
+---
+type: docs
+title: "Sentinel"
+linkTitle: "Sentinel"
+weight: 7000
+description: "Use Sentinel middleware to guarantee the reliability and resiliency of your application"
+---
+
+[Sentinel](https://github.com/alibaba/sentinel-golang) is a powerful fault-tolerance component that takes "flow" as the breakthrough point and covers multiple fields including flow control, traffic shaping, concurrency limiting, circuit breaking, and adaptive system protection to guarantee the reliability and resiliency of microservices.
+
+The Sentinel [HTTP middleware]({{< ref middleware-concept.md >}}) enables Dapr to facilitate Sentinel's powerful abilities to protect your application. You can refer to [Sentinel Wiki](https://github.com/alibaba/sentinel-golang/wiki) for more details on Sentinel.
+
+## Component format
+
+In the following definition, the maximum requests per second are set to 10:
+
+```yaml
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: sentinel
+spec:
+  type: middleware.http.sentinel
+  version: v1
+  metadata:
+  - name: appName
+    value: "nodeapp"
+  - name: logDir
+    value: "/var/tmp"
+  - name: flowRules
+    value: >-
+      [
+        {
+          "resource": "POST:/v1.0/invoke/nodeapp/method/neworder",
+          "threshold": 10,
+          "tokenCalculateStrategy": 0,
+          "controlBehavior": 0
+        }
+      ]
+```
+
+## Spec metadata fields
+
+| Field       | Details                                                                                                                                                                              | Example |
+|----------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------|
+| appName | the name of current running service | `nodeapp`    |
+| logDir | the log directory path | `/var/tmp/sentinel` |
```

Comment:
Does this work on Kubernetes?

Reply:
yes, it's a general solution and is supposed to work in k8s environment too.